### PR TITLE
feat(star_fusion): rename outfile

### PR DESCRIPTION
**Pull request by jemten**
### This PR adds | fixes:

- Renames star fusion outfile so that it contains the sample id

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by jemten** [comment]
@henrikstranneheim, @ramprasadn bump

File: lib/MIP/Recipes/Analysis/Star_fusion.pm [hunk 1]
```diff
@@ -165,9 +165,14 @@ sub analysis_star_fusion {
             recipe_name           => $recipe_name,
         }
     );
-    my $outdir_path    = catdir( $active_parameter_href->{outdata_dir}, $sample_id, $recipe_name );
-    my $outsample_name = $STAR_FUSION_PREFIX . $recipe{outfile_suffix};
-    my @file_paths     = catfile( $outdir_path, $outsample_name );
+
+    ## Build outfile_paths
+    my $outdir_path = catdir( $active_parameter_href->{outdata_dir}, $sample_id, $recipe_name );
+    my $lanes_id    = join $EMPTY_STR, @{ $file_info_href->{$sample_id}{lanes} };
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Star_fusion.pm:171]: I think we should add a get_sample_id_lanes sub here as we are traversing 2 levels. Maybe there is a general sub one can reuse or extend but I am not certain

File: lib/MIP/Recipes/Analysis/Star_fusion.pm [hunk 2]
```diff
@@ -165,9 +165,14 @@ sub analysis_star_fusion {
...
+    my $outfile_tag         = $file_info_href->{$sample_id}{$recipe_name}{file_tag};
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Analysis/Star_fusion.pm:172]: I think we should use a get sub here as well

**Review by henrikstranneheim — APPROVED** [review]
👍

File: lib/MIP/Recipes/Analysis/Star_fusion.pm (hunk 1 above)
Comment (jemten) [reply lib/MIP/Recipes/Analysis/Star_fusion.pm:171]: Yeah, I can fix a sub for that

**Comment by jemten** [comment]
A bit better, right. parse_io_outfiles could take care of some of the complexity

File: lib/MIP/File_info.pm [hunk 3]
```diff
@@ -34,6 +34,7 @@ BEGIN {
       get_is_sample_files_compressed
       get_merged_infile_prefix
       get_sample_file_attribute
+      get_sample_fastq_file_lanes
```
Comment (henrikstranneheim) [inline lib/MIP/File_info.pm:37]: Order AB

File: lib/MIP/File_info.pm [hunk 4]
```diff
@@ -685,6 +686,40 @@ sub get_sample_file_attribute {
     return $stored_attribute;
 }
 
+sub get_sample_fastq_file_lanes {
+
+## Function : Get sample fastq filelanes
```
Comment (henrikstranneheim) [inline lib/MIP/File_info.pm:691]:
```suggestion
## Function : Get sample fastq file lanes
```

**Review by henrikstranneheim — APPROVED** [review]
Looks great! 👍

**Pull request merged**